### PR TITLE
Add color picker

### PR DESCRIPTION
**Pull request by thomas771020**

### CODE REVIEW

**Comment by john-hu** [comment]
This looks good. But it still has some spaces to improve it. Please rebase the code and let user to choose a color non-sequential.

**Comment by thomas771020** [comment]
Hi, John,  I have already updated this pull request. Since the wheel color seem more difficult to use, I decide use a liner color selector. Please help me check this is ok or not. Thank you~

File: game/js/game.js [hunk 1]
```diff
@@ -82,15 +82,82 @@ $(function docReady() {
...
-      idx = 0;
+  function genGradualColors(color, colorMap){
+    var arr = [];
+    for(var i = 0; i < colorMap.length; i++){
+      if(color === 'r'){
+        arr.push('rgb(' + colorMap[i] + ', 0, 0)');
+      } else if(color === 'g'){
+        arr.push('rgb(0, ' + colorMap[i] + ', 0)');
+      }
+      else if(color === 'b'){
```
Comment (john-hu) [inline game/js/game.js]: Please move line 93 to line 92

File: game/js/game.js [hunk 2]
```diff
@@ -82,15 +82,82 @@ $(function docReady() {
     $('.mixed-color').css('background-color', 'rgb(' + r +', ' + g + ', ' + b + ')');
   }
 
-  function handleColorClicked(e) {
-    var idx = $(e.target).data('color-index');
-    if (++idx >= level) {
-      idx = 0;
+  function genGradualColors(color, colorMap){
```
Comment (john-hu) [inline game/js/game.js]: Please add a white space between `)` and `{`

File: game/js/game.js [hunk 3]
```diff
@@ -82,15 +82,82 @@ $(function docReady() {
...
+    var arr = [];
+    for(var i = 0; i < colorMap.length; i++){
```
Comment (john-hu) [inline game/js/game.js]: Please add a white space between `)` and `{`, and `for` and `(`...

File: game/js/game.js [hunk 4]
```diff
@@ -82,15 +82,82 @@ $(function docReady() {
...
+      if(color === 'r'){
```
Comment (john-hu) [inline game/js/game.js]: Please add a white space between `)` and `{`

File: game/js/game.js [hunk 5]
```diff
@@ -82,15 +82,82 @@ $(function docReady() {
...
+        arr.push('rgb(' + colorMap[i] + ', 0, 0)');
+      } else if(color === 'g'){
```
Comment (john-hu) [inline game/js/game.js]: Please add a white space between `)` and `{`

File: game/js/game.js [hunk 6]
```diff
@@ -82,15 +82,82 @@ $(function docReady() {
...
+        arr.push('rgb(0, 0, ' + colorMap[i] + ')');
+      }
     }
-    $(e.target).data('color-index', idx);
+    return arr;
+  }
 
-    renderColor();
-    checkAnswer();
+  function ModalColorPicker() {
```
Comment (john-hu) [inline game/js/game.js]: Please move ModalColorPicker as a standalone file instead of making it as inline...

**Comment by john-hu** [comment]
@thomas771020 Thank you for this version. It looks pretty good. Please revise the code and send it back. It would be nice to give me a screenshot of this one. I will merge for you after revised.

**Comment by thomas771020** [comment]
This is the screenshot for ModalColorPicker.

**Pull request merged**